dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/sophon/gas_sophon_fees.sql

```diff
@@ -0,0 +1,97 @@
+{% set blockchain = 'sophon' %}
+
+{{ config(
+    schema = 'gas_' + blockchain
+    ,alias = 'fees'
+    ,partition_by = ['block_month']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy='merge'
+    ,unique_key = ['block_month', 'tx_hash']
+    ,incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set test_short_ci = false %}
+
+WITH base_model as (
+    SELECT
+        txns.block_time
+        ,txns.block_number
+        ,txns.hash AS tx_hash
+        ,txns."from" AS tx_from
+        ,txns.to AS tx_to
+        ,cast(gas_price as uint256) as gas_price
+        ,txns.gas_used as gas_used
+        ,cast(gas_price as uint256) * cast(txns.gas_used as uint256) as tx_fee_raw
+        ,case when txns.priority_fee_per_gas is null or base_fee_per_gas = gas_price or txns.priority_fee_per_gas < 0
+            then map(array['base_fee'], array[cast(gas_price as uint256) * cast(txns.gas_used as uint256)])
+            else map(
+                array['base_fee', 'priority_fee'],
+                array[
+                    cast(base_fee_per_gas as uint256) * cast(txns.gas_used as uint256),
+                    cast(priority_fee_per_gas as uint256) * cast(txns.gas_used as uint256)
+                ]
+            )
+        end as tx_fee_breakdown_raw
+        ,{{var('ETH_ERC20_ADDRESS')}} as tx_fee_currency
```

Comment:
```suggestion
        , 0x000000000000000000000000000000000000800a as tx_fee_currency
```

Reply:
plz always cross-reference the new source of truth
```
select *
from dune.blockchains
where name = 'sophon'
```

Reply:
we will likely eventually replace the logic here to inner join to that table and grab address, but not at this moment until we roll out spellbook globally